main/src/main/java/cgeo/geocaching/storage/DataStore.java

```diff
@@ -4324,6 +4327,29 @@ private static void deleteOrphanedUDC() {
         removeCaches(orphanedUDC, LoadFlags.REMOVE_ALL);
     }
 
+    private static void deleteOrphanedTrackfiles() {
+        // current used trackfiles
+        final ArrayList<Trackfiles> currentTrackFiles = Trackfiles.getTrackfiles();
+        // all trackfiles
+        final List<ImmutablePair<ContentStorage.FileInformation, String>> trackFiles = FolderUtils.get().getAllFiles(Folder.fromFile(LocalStorage.getTrackfilesDir()));
```

Comment:
You are using the ContentStorage framework here. Although this will work, you may make your life more complex than necessary. Because the trackfile copies are stored in app internal storage, they will always be referencable using simple `File` operations instead.

E.g. here: 
`final FIle[] trackFiles = LocalStorage.getTrackFilesDir().listFiles()`

And then later to delete one of them:
`trackFiles[i].delete()`

(I am assuming that all track files are stored in same dir and that there is no subdirectory structure beneath `LocalStorage.getTrackFilesDir()`